baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/ConsistentOverrides.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.errorprone;
+
+import com.google.auto.service.AutoService;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.BugPattern.SeverityLevel;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.bugpatterns.BugChecker.MethodTreeMatcher;
+import com.google.errorprone.fixes.SuggestedFixes;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.source.tree.MethodTree;
+import com.sun.source.tree.VariableTree;
+import com.sun.tools.javac.code.Symbol;
+import com.sun.tools.javac.code.Symbol.MethodSymbol;
+import com.sun.tools.javac.code.Symbol.VarSymbol;
+import com.sun.tools.javac.code.Type;
+import com.sun.tools.javac.code.Types;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import org.immutables.value.Value.Immutable;
+
+@AutoService(BugChecker.class)
+@BugPattern(
+        name = "ConsistentOverrides",
+        link = "https://github.com/palantir/gradle-baseline#baseline-error-prone-checks",
+        linkType = BugPattern.LinkType.CUSTOM,
+        providesFix = BugPattern.ProvidesFix.REQUIRES_HUMAN_ATTENTION,
+        severity = SeverityLevel.ERROR,
+        summary = "Method overrides should have variable names consistent with the super-method when there "
+                + "are multiple parameters with the same type to avoid incorrectly binding values to variables.")
+public final class ConsistentOverrides extends BugChecker implements MethodTreeMatcher {
+    private static final Pattern UNKNOWN_ARG_NAME = Pattern.compile("^args\\d+$");
+
+    @Override
+    public Description matchMethod(MethodTree tree, VisitorState state) {
+        Symbol sym = ASTHelpers.getSymbol(tree);
+        if (!(sym instanceof MethodSymbol) || sym.isStatic()) {
+            return Description.NO_MATCH;
+        }
+
+        MethodSymbol methodSymbol = (MethodSymbol) sym;
+        List<? extends VariableTree> paramTrees = tree.getParameters();
+        getNonParameterizedSuperMethod(methodSymbol, state.getTypes())
+                .filter(ConsistentOverrides::hasMeaningfulArgNames)
```

Comment:
I'm not sure that's exactly what you want though since it would let terrible names slip by the errorprone rule. I think we should assume that the local build has been configured "correctly" since if it hasn't the user still has agency to fix it

Reply:
The error message won't give them anything to work with though. If their names are correct, the'll get something along the lines of "You wrote 'void foo(int first, int second)', did you mean 'void foo(int first, int second)'?

Enforcing that the feature is enabled is reasonable, but I don't think this check should be responsible for that enforcement.

Reply:
ok, lets follow up with something in baseline to ensure that `-parameters` is enabled?

Reply:
Sounds good, #1690 handles most folks who don't opt into individual plugins, but we don't have anything to warn stragglers to enable the plugin.